src/decorators/index.ts

```diff
@@ -24,22 +24,21 @@ import { loggerFor } from '../logger';
 import { config } from '../config';
 const L = loggerFor(__filename);
 
-// TODO: MAKE ENV VARIABLE
-const EGO_URL = 'https://ego.dev.argo.cancercollaboratory.org/api/';
-
 const getToken = async (request: Request) => {
   if (!request.headers.authorization) {
     return undefined;
   }
 
-  const token = request.headers.authorization.includes('Bearer')
-    ? decodeAndVerify(request.headers.authorization.split(' ')[1])
-    : await verifyEgoApiKey(request.headers.authorization, request.headers.authorization);
+  const authToken = request.headers.authorization.split(' ')[1];
+
+  const token = Boolean(jwt.decode(authToken))
+    ? verifyJwt(authToken)
+    : await verifyEgoApiKey(authToken);
```

Comment:
We still need a check somewhere that this header starts with "Bearer"
```suggestion
  if (!request.headers.authorization.startsWith('Bearer ') {
    return undefined;
  }

  const authToken = request.headers.authorization.slice(7); // remove 'Bearer ' from header

  const token = Boolean(jwt.decode(authToken))
    ? verifyJwt(authToken)
    : await verifyEgoApiKey(authToken);
```

Reply:
I would actually add a trim to line 36 here to deal with extra whitespace issues:
` const authToken = request.headers.authorization.slice(7).trim();`

Reply:
Updated